pynamodb/connection/base.py

```diff
@@ -95,6 +95,23 @@ def hash_keyname(self):
                     break
         return self._hash_keyname
 
+    def get_key_names(self, index_name=None):
+        """
+        Returns the names of the primary key attributes and index key attributes (if index_name is specified)
+        """
+        key_names = [self.hash_keyname]
+        range_keyname = self.range_keyname
```

Comment:
```python
if self.range_keyname:
    key_names.append(self.range_keyname)
```

Reply:
that would require us to search for it twice if it doesn't exist

Reply:
search? twice? it's only used once in this method.

Reply:
self.range_keyname is a property computed by scanning the metadata -- its memoized, but only if the value is not `None` -- if it is `None`, the if statement will search and then ....

actually you're right, if it's None you won't enter the if block -- updating